cloud-reactor/src/test/java/com/sequenceiq/cloudbreak/cloud/handler/ProvisionValidationHandlerTest.java

```diff
@@ -0,0 +1,134 @@
+package com.sequenceiq.cloudbreak.cloud.handler;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.cloudbreak.cloud.Authenticator;
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.Validator;
+import com.sequenceiq.cloudbreak.cloud.ValidatorType;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.event.setup.ValidationRequest;
+import com.sequenceiq.cloudbreak.cloud.event.setup.ValidationResult;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudPlatformValidationWarningException;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.CloudStack;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@ExtendWith(MockitoExtension.class)
+public class ProvisionValidationHandlerTest {
+
+    private static final long STACK_ID = 1L;
+
+    private static final String WARNING_MESSAGE = "warning";
+
+    @InjectMocks
+    private ProvisionValidationHandler underTest;
+
+    @Mock
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Mock
+    private EventBus eventBus;
+
+    @Mock
+    private CloudConnector<Object> cloudConnector;
+
+    @Mock
+    private Authenticator authenticator;
+
+    @Mock
+    private AuthenticatedContext ac;
+
+    @Mock
+    private Validator validator;
+
+    @Test
+    void testAcceptShouldSendEventWhenAllValidationWereSuccessful() {
+        ValidationRequest request = createValidationRequest();
+        CloudContext cloudContext = request.getCloudContext();
+        mockCloudConnector(request, cloudContext);
+
+        underTest.accept(new Event<>(request));
+
+        verifyCloudConnectorCalls(request, cloudContext);
+        ArgumentCaptor<Event> argumentCaptor = ArgumentCaptor.forClass(Event.class);
+        verify(eventBus).notify(eq("VALIDATIONRESULT"), argumentCaptor.capture());
+        Event<ValidationResult> resultEvent = argumentCaptor.getValue();
+        assertTrue(resultEvent.getData().getWarningMessages().isEmpty());
+    }
+
+    @Test
+    void testAcceptShouldSendEventWhenAllValidationWereSuccessfulWhenTheValidatorThrowsWarningMessage() {
```

Comment:
I would make this name shorter and leave part  "WhenAllValidationWereSuccessful"